components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/UserProfileAdmin.java

```diff
@@ -568,8 +567,7 @@ public void associateID(String idpID, String associatedID) throws UserProfileExc
         PreparedStatement prepStmt = null;
         String sql = null;
         int tenantID = CarbonContext.getThreadLocalCarbonContext().getTenantId();
-        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(CarbonContext.getThreadLocalCarbonContext()
-                                                                          .getUsername());
+        String tenantAwareUsername = CarbonContext.getThreadLocalCarbonContext().getUsername();
         String domainName = getDomainName(tenantAwareUsername);
         tenantAwareUsername = getUsernameWithoutDomain(tenantAwareUsername);
```

Comment:
Get rid of private method getUsernameWithoutDomain() and use UserCoreUtil.removeDomainFromName() method